validator/core/src/input/proto.rs

```diff
@@ -0,0 +1,205 @@
+//! Module for representing Substrait protobuf input.
+//!
+//! The structures here are generated using [`prost`], but have a bunch of
+//! extra traits from [`traits`](crate::input::traits) associated with them,
+//! for which the implementations are generated using
+//! [`substrait_validator_derive`]. The purpose of these traits is to add basic
+//! introspection capabilities to the prost structures. One of the use cases
+//! for this is to let the parsing code automatically detect when the
+//! validation code ignored a subtree while validating, which implies that the
+//! validator hasn't checked everything and thus should not warrant that the
+//! received plan is valid.
+
+use crate::input::traits;
+use crate::output::primitive_data;
+
+#[allow(clippy::large_enum_variant)]
+pub mod substrait {
+    include!(concat!(env!("OUT_DIR"), "/substrait.rs"));
+    pub mod extensions {
+        include!(concat!(env!("OUT_DIR"), "/substrait.extensions.rs"));
+    }
+    pub mod validator {
+        include!(concat!(env!("OUT_DIR"), "/substrait.validator.rs"));
+    }
+}
+
+impl traits::ProtoPrimitive for bool {
+    fn proto_primitive_type() -> &'static str {
+        "bool"
+    }
+
+    fn proto_primitive_default() -> primitive_data::PrimitiveData {
+        primitive_data::PrimitiveData::Bool(false)
+    }
+
+    fn proto_primitive_data(&self) -> primitive_data::PrimitiveData {
+        primitive_data::PrimitiveData::Bool(*self)
+    }
+
+    fn proto_primitive_is_default(&self) -> bool {
+        !*self
```

Comment:
```suggestion
        *self == false
```

Reply:
I actually have to revert this because clippy doesn't agree with you.